File: sleepy-test/java-junit/ArchiveFileNameHelper_DeleteArchiveFilesByDate_AlteredMaxArchive.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.*;
import java.util.stream.Collectors;

public class ArchiveFileNameHelper {

    @Test
    public void deleteArchiveFilesByDate_AlteredMaxArchive() throws Exception {
        Path tempPath = Paths.get(System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString());
        Path logFile = tempPath.resolve("file.txt");

        try {
            Path archiveFolder = tempPath.resolve("archive");
            FileTarget fileTarget = new FileTarget();
            fileTarget.setFileName(logFile.toString());
            fileTarget.setArchiveFileName(archiveFolder.resolve("{#}.txt").toString());
            fileTarget.setArchiveAboveSize(50);
            fileTarget.setLineEnding(LineEndingMode.LF);
            fileTarget.setArchiveNumbering(ArchiveNumberingMode.DATE);
            fileTarget.setArchiveDateFormat("yyyyMMddHHmmssSSS");
            fileTarget.setLayout("${message}");
            fileTarget.setMaxArchiveFiles(5);

            SimpleConfigurator.configureForTargetLogging(fileTarget, LogLevel.DEBUG);

            for (int i = 0; i < 29; ++i) {
                logger.debug("123456789");
                if (i % 5 == 0) {
                    Thread.sleep(50);
                }
            }

            LogManager.setConfiguration(null);
            List<Path> files = Files.list(archiveFolder).sorted().collect(Collectors.toList());
            assertEquals(fileTarget.getMaxArchiveFiles(), files.size());

            fileTarget.setMaxArchiveFiles(2);
            SimpleConfigurator.configureForTargetLogging(fileTarget, LogLevel.DEBUG);
            logger.debug("1234567890");
            LogManager.setConfiguration(null);

            List<Path> files2 = Files.list(archiveFolder).sorted().collect(Collectors.toList());
            assertEquals(fileTarget.getMaxArchiveFiles(), files2.size());
            assertFalse(files2.contains(files.get(0)));
            assertFalse(files2.contains(files.get(1)));
            assertFalse(files2.contains(files.get(2)));
            assertFalse(files2.contains(files.get(3)));
            assertEquals(files.get(4), files2.get(0));
            assertFalse(files.contains(files2.get(1)));
        } finally {
            Files.deleteIfExists(logFile);
            if (Files.exists(tempPath)) {
                Files.walk(tempPath)
                    .sorted(Comparator.reverseOrder())
                    .map(Path::toFile)
                    .forEach(File::delete);
            }
        }
    }
}

File: sleepy-test/python-unittest/ArchiveFileNameHelper_DeleteArchiveFilesByDate_AlteredMaxArchive.py
import unittest
import os
import time
import shutil
from pathlib import Path

class TestArchiveFileNameHelper(unittest.TestCase):
    def test_delete_archive_files_by_date_altered_max_archive(self):
        temp_path = os.path.join(Path(os.getenv("TEMP") or "/tmp"), str(uuid.uuid4()))
        log_file = os.path.join(temp_path, "file.txt")

        try:
            archive_folder = os.path.join(temp_path, "archive")
            file_target = FileTarget(
                file_name=log_file,
                archive_file_name=os.path.join(archive_folder, "{#}.txt"),
                archive_above_size=50,
                line_ending=LineEndingMode.LF,
                archive_numbering=ArchiveNumberingMode.DATE,
                archive_date_format="yyyyMMddHHmmssfff",
                layout="${message}",
                max_archive_files=5,
            )

            SimpleConfigurator.configure_for_target_logging(file_target, LogLevel.DEBUG)

            for i in range(29):
                logger.debug("123456789")
                if i % 5 == 0:
                    time.sleep(0.05)

            LogManager.configuration = None
            files = sorted(os.listdir(archive_folder))
            self.assertEqual(file_target.max_archive_files, len(files))

            file_target.max_archive_files = 2
            SimpleConfigurator.configure_for_target_logging(file_target, LogLevel.DEBUG)
            logger.debug("1234567890")
            LogManager.configuration = None
            files2 = sorted(os.listdir(archive_folder))

            self.assertEqual(file_target.max_archive_files, len(files2))
            self.assertNotIn(files[0], files2)
            self.assertNotIn(files[1], files2)
            self.assertNotIn(files[2], files2)
            self.assertNotIn(files[3], files2)
            self.assertEqual(files[4], files2[0])
            self.assertNotIn(files2[1], files)
        finally:
            if os.path.exists(log_file):
                os.remove(log_file)
            if os.path.exists(temp_path):
                shutil.rmtree(temp_path)
